ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/AbstractDebugTraceBlock.java

```diff
@@ -0,0 +1,157 @@
+      final JsonRpcRequestContext requestContext,
+      final TraceOptions traceOptions,
+      final Optional<Block> maybeBlock) {
+    return maybeBlock
+        .flatMap(
+            block ->
+                Tracer.processTracing(
+                    getBlockchainQueries(),
+                    block.getHash(),
+                    traceableState -> {
+                      Collection<DebugTraceTransactionResult> tracesList =
+                          new CopyOnWriteArrayList<>();
+                      final ProtocolSpec protocolSpec =
+                          protocolSchedule.getByBlockHeader(block.getHeader());
+                      final MainnetTransactionProcessor transactionProcessor =
+                          protocolSpec.getTransactionProcessor();
+                      final TraceBlock.ChainUpdater chainUpdater =
+                          new TraceBlock.ChainUpdater(traceableState);
+
+                      TransactionSource transactionSource = new TransactionSource(block);
+                      DebugOperationTracer debugOperationTracer =
+                          new DebugOperationTracer(traceOptions, true);
+                      ExecuteTransactionStep executeTransactionStep =
+                          new ExecuteTransactionStep(
+                              chainUpdater,
+                              transactionProcessor,
+                              getBlockchainQueries().getBlockchain(),
+                              debugOperationTracer,
+                              protocolSpec,
+                              block);
+                      DebugTraceTransactionStep debugTraceTransactionStep =
+                          new DebugTraceTransactionStep();
+                      Pipeline<TransactionTrace> traceBlockPipeline =
+                          createPipelineFrom(
+                                  "getTransactions",
+                                  transactionSource,
+                                  4,
+                                  outputCounter,
+                                  false,
+                                  "debug_trace_block")
+                              .thenProcess("executeTransaction", executeTransactionStep)
+                              .thenProcessAsyncOrdered(
+                                  "debugTraceTransactionStep", debugTraceTransactionStep, 4)
+                              .andFinishWith("collect_results", tracesList::add);
+
+                      try {
+                        if (getBlockchainQueries().getEthScheduler().isPresent()) {
+                          getBlockchainQueries()
+                              .getEthScheduler()
+                              .get()
+                              .startPipeline(traceBlockPipeline)
+                              .get();
+                        } else {
+                          EthScheduler ethScheduler =
+                              new EthScheduler(1, 1, 1, 1, new NoOpMetricsSystem());
```

Comment:
This is a little mysterious, might be good to name the variable something like "singleThreadedEthScheduler" perhaps? Don't we want metrics too?

Reply:
I changed the implementation and deleted this code https://github.com/hyperledger/besu/commit/6a541e5798cccfe8bcc580658c85f8809599ea03